pkg/tsdb/cloudwatch/request_parser_test.go

```diff
@@ -363,26 +386,35 @@ func Test_migrateAliasToDynamicLabel_single_query_preserves_old_alias_and_create
 						"statistic": "Average",
 						"period": "600",
 						"hide": false
-				  }`, tc.inputAlias)))
+				  }`, tc.inputAlias))
+
+			var query QueryJson
+			err := json.Unmarshal(queryJson, &query)
 			require.NoError(t, err)
 
-			migrateAliasToDynamicLabel(queryJson)
+			migrateAliasToDynamicLabel(&query)
 
-			assert.Equal(t, simplejson.NewFromAny(
-				map[string]interface{}{
-					"alias":      tc.inputAlias,
-					"dimensions": map[string]interface{}{"InstanceId": []interface{}{"test"}},
-					"hide":       false,
-					"label":      tc.expectedLabel,
-					"metricName": "CPUUtilization",
-					"namespace":  "ec2",
-					"period":     "600",
-					"region":     "us-east-1",
-					"statistic":  "Average"}), queryJson)
+			matchedJson := []byte(fmt.Sprintf(`{
+				"alias":      "%s",
+				"dimensions": {
+					"InstanceId": ["test"]
+				},			
+				"hide":       false,
+				"label":      "%s",
+				"metricName": "CPUUtilization",
+				"namespace":  "ec2",
+				"period":     "600",
+				"region":     "us-east-1",
+				"statistic":  "Average"
+			}`, tc.inputAlias, tc.expectedLabel))
+
+			result, err := json.Marshal(query)
+			require.NoError(t, err)
+
+			assert.JSONEq(t, string(matchedJson), string(result))
```

Comment:
This is asserted the expected (result) against the expected (matchedJson).

It should assert against the actual (query) since it is modified in place.

Reply:
Sorry I've misread this!